samples/04.0-file-transfer/transfer-file/src/main/java/org/eclipse/dataspaceconnector/extensions/api/FileTransferExtension.java

```diff
@@ -40,23 +36,23 @@
     public static final String USE_POLICY = "use-eu";
     private static final String EDC_ASSET_PATH = "edc.samples.04.asset.path";
     @Inject
-    private DataFlowManager dataFlowMgr;
-    @Inject
-    private DataAddressResolver dataAddressResolver;
-    @Inject
-    private DataOperatorRegistry dataOperatorRegistry;
-    @Inject
     private ContractDefinitionStore contractStore;
     @Inject
     private AssetLoader loader;
+    @Inject
+    private PipelineService pipelineService;
+    @Inject
+    private DataTransferExecutorServiceContainer executorContainer;
 
     @Override
     public void initialize(ServiceExtensionContext context) {
-        var vault = context.getService(Vault.class);
-        dataOperatorRegistry.registerStreamPublisher(new FileTransferDataStreamPublisher(context.getMonitor(), dataAddressResolver));
+        var monitor = context.getMonitor();
+
+        var sourceFactory = new FileTransferDataSourceFactory();
+        pipelineService.registerFactory(sourceFactory);
 
-        DataFlowController dataFlowController = new InlineDataFlowController(vault, context.getMonitor(), dataOperatorRegistry, dataAddressResolver);
```

Comment:
When are we going to remove the `InlineDataFlowController`? In the subsequent story when we update the sample?

Reply:
It is still used under Sample05. May be we can mark `InlineDataFlowController` as @Deprecated, what do you think @algattik @marcgs ?

Reply:
Sound good to me. Perhaps with a reference to the issue that will do the work?

Reply:
I will create an upstream issue after opening these changes for EDC review.

Reply:
created https://github.com/eclipse-dataspaceconnector/DataSpaceConnector/issues/1061